lib/json/JsonParser.js

```diff
@@ -53,6 +59,31 @@ class JsonParser extends Parser {
 		} catch (e) {
 			throw new Error(`Cannot parse JSON: ${/** @type {Error} */ (e).message}`);
 		}
+
+		// If the module is associated with a layer, try to reuse cached data instead
+		// of duplicating the data multiple times.
+		const module = state.module;
+		if (module && module.userRequest && module.layer && Buffer.isBuffer(data)) {
+			let moduleLayerCache = cachedModuleLayerCaches.get(
+				this.associatedObjectForCache
+			);
+			if (moduleLayerCache) {
+				moduleLayerCache = new ModuleLayerCache();
+				cachedModuleLayerCaches.set(
+					this.associatedObjectForCache,
+					moduleLayerCache
+				);
+			}
+			if (moduleLayerCache.has(module.userRequest)) {
+				const cachedData = moduleLayerCache.get(module.userRequest);
+				if (Buffer.isBuffer(cachedData) && data.equals(cachedData)) {
+					data = cachedData;
+				}
+			} else {
+				moduleLayerCache.set(module.userRequest, data);
+			}
+		}
```

Comment:
Can we move it to method, to avoid duplicate logic (for example to the compilation class)

Reply:
Done. Sorry for the delay. Let me know if this is what you had in mind.